contracts/DecentralizedKV.sol

```diff
@@ -104,16 +104,11 @@ contract DecentralizedKV {
             lastKvIdx = lastKvIdx + 1;
         }
         paddr.kvSize = uint24(data.length);
-        paddr.hash = bytes24(keccak256(data));
+        uint256 nChunkBits = generateChunkBits(data.length);
+        paddr.hash = MerkleLib.merkleRoot(data, SYSTEM_BLOCK_SIZE, nChunkBits);
         kvMap[skey] = paddr;
 
-        // Weird that cannot call precompiled contract like this (solidity issue?)
-        // storageManager.putRaw(paddr.kvIdx, data);
-        // Use call directly instead.
-        (bool success, ) = address(storageManager).call(
-            abi.encodeWithSelector(IStorageManager.putRaw.selector, paddr.kvIdx, data)
-        );
-        require(success, "failed to putRaw");
+        storageManager.putRaw(paddr.kvIdx, data);
```

Comment:
I would recommend keeping the original code as we found an issue when running on the actual network.